internal/controller/etcdcluster_controller.go

```diff
@@ -132,177 +131,24 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 func (r *EtcdClusterReconciler) ensureClusterObjects(
 	ctx context.Context, cluster *etcdaenixiov1alpha1.EtcdCluster, isClusterInitialized bool) error {
 	// 1. create or update configmap <name>-cluster-state
-	if err := r.ensureClusterStateConfigMap(ctx, cluster, isClusterInitialized); err != nil {
+	if err := factory.CreateOrUpdateClusterStateConfigMap(ctx, cluster, isClusterInitialized, r.Client, r.Scheme); err != nil {
```

Comment:
Not a blocker, just slight concerns about the naming: despite of the argument name "isClusterInitialized", the real value which is passed to function reflects the rediness state of the cluster (https://github.com/aenix-io/etcd-operator/blob/feat/refactor-resource-generation/internal/controller/etcdcluster_controller.go#L87-L100)

```go
	isClusterReady := false
	sts := &appsv1.StatefulSet{}
	err = r.Get(ctx, client.ObjectKey{
		Namespace: instance.Namespace,
		Name:      instance.Name,
	}, sts)
	if err == nil {
		isClusterReady = sts.Status.ReadyReplicas == *sts.Spec.Replicas
	}

	if err := r.ensureClusterObjects(ctx, instance, isClusterReady); err != nil {
		logger.Error(err, "cannot create Cluster auxiliary objects")
		return r.updateStatusOnErr(ctx, instance, fmt.Errorf("cannot create Cluster auxiliary objects: %w", err))
	}
```
Which might lead to misunderstanding and confusion

Reply:
Updated naming